util/logtools/generate_triggers.ts

```diff
@@ -0,0 +1,1386 @@
+            castTypeFullSuggestions.add('Circle (out)');
+            break;
+          case 3: // Cone, size modified by hitbox
+          case 13: // Cone
+            castTypeSuggestions.add('Away from Front (cone)');
+            castTypeFullSuggestions.add('Away from Front (cone)');
+            break;
+          case 4: // Rectangle, size modified by hitbox
+          case 12: // Rectangle
+          case 8: // Charging rectangle
+            castTypeSuggestions.add('Custom Separate');
+            castTypeFullSuggestions.add(`Rectangle AoE, CastType = ${abilityInfo.fields.CastType}`);
+            break;
+          case 10: // Donut
+            castTypeSuggestions.add('Donut (in)');
+            castTypeFullSuggestions.add('Donut (in)');
+            break;
+          case 11: // Plus
+            castTypeSuggestions.add('Custom Joined');
+            castTypeFullSuggestions.add(`Plus AoE`);
+            break;
+        }
+      }
+    }
+
+    if (castTypeSuggestions.size !== 0) {
+      // Just apply the first entry in the set
+      suggestedOperation = [...castTypeSuggestions][0] ?? 'Skip';
+    } else {
+      if (abilityLineTypes.includes('22') && abilityLineTypes.includes('21')) {
+        // Sometimes hits, sometimes doesn't, this is probably a dodgeable mechanic
+        suggestedOperation = 'Custom Joined';
+      } else if (abilityLineTypes.includes('21') && hitAPlayer) {
+        // Never hits more than 1 person, but has hit a person
+        // Also has a cast bar, this is probably a dodgeable mechanic
+        suggestedOperation = 'Custom Joined';
+      } else if (abilityLineTypes.includes('21')) {
+        // Has a cast bar, finished casting, never hit a player.
+        suggestedOperation = 'Skip';
+      } else {
+        // Cast, and always hits multiple
+        suggestedOperation = 'AoE';
+      }
+    }
+
+    const result = await inquirer.prompt<{ action: TriggerSuggestTypes | null }>([
+      {
+        type: 'list',
+        name: 'action',
+        message: `Ability Information:
+Name: ${abilityName},
+IDs: ${mapInfo.ids.join(', ')},
+Sources: ${
+          [
+            ...new Set(mapInfo.fights.flatMap((fight) =>
+              fight.instances.map((instance) => instance.groups?.source ?? '')
+            )),
+          ].sort().join(', ')
+        },
+Line Types: ${abilityLineTypes.sort(numberSort).join(', ')},
+Line Count: ${instances.length},
+Offsets: ${mapInfo.offsets.sort(numberSort).join(', ')},
+CastInfo Hints: ${[...castTypeFullSuggestions].join(', ')}
+`,
+        choices: triggerSuggestOptions.map((e) => {
+          return {
+            name: e,
+            value: e,
+          };
+        }),
+        default: suggestedOperation,
+      },
+    ]);
+
+    switch (result.action) {
+      case 'AoE':
+        triggersText += `
+    {
+      id: '${args.trigger_id_prefix ?? ''} ${abilityName}',
+      type: 'StartsUsing',
+      netRegex: { id: '${mapInfo.ids.join('|').toUpperCase()}', source: '${
```

Comment:
```suggestion
      netRegex: { id: '[${mapInfo.ids.join(',').toUpperCase()}]', source: '${
```
Here and elsewhere. To date, for multiple values in any netRegex param, we've typically formatted those values as an array rather than pipe-separated. (Does pipe separation without enclosing parens even work? I've never tried it.)

(This is a drive-by, I'll take a better look later when there's a bit more time.)

Reply:
This suggestion would have produced invalid TS/JS (`id: '[ABCD, DEFG]', source...`). I solved it properly I believe in the latest commit.